solution -- background queue

antdke opened the pull request:
hey @ryanckulp, this is my submission for issue #3 

i built off my code i submitted for issue #4 . i know that code hasn't been reviewed/approved/improved by you yet but i figured that i could still work on issue #3 in the meantime and then incorporate your improvements to my code in issue #4 then re-commit the code in this branch for this issue.

hopefully that's ok!

everything here works:

1. Implemented [Betterment's delayed](https://github.com/betterment/delayed).  And now all TwitterService invocations run in the background, with a 1 minute delay between each job.
2. After submitting a feedback, users see an estimate of when their feedback will be tweeted by our bot account. I also redirected users to a new page w/o the form so that there's more of an "emphasis" that they have to wait for their feedback to be tweeted and to deter users from submitting many feedbacks consecutively.
3. Made the delay (e.g. 1 minute) easily modifiable via an ENV variable: `delayed_delay_minutes`

lmk your thoughts! :)

(btw, learning so much working on these issues. the bounty is a great incentive. especially since i'm unemployed haha 🥲)

antdke commented:
@ryanckulp 

this branch should be good to merge :)

i merged in the improvements you made to #12 

only thing I added here was an ENV variable to `application-sample.yml` for the delay minutes value (`delayed_delay_minutes`), so that they developer can modify the time that tweets are delayed in a queue.

lmk your thoughts!

ryanckulp commented:
@antdke thanks for the PR!

i think this can be significantly simplified via 1 of 2 strategies:

1. refactor your delayed job `wait_until` to simply be something like `wait_until: (Delayed::Job.count + 1).minutes.later)` to send the tweet in the appropriate number of minutes from now
2. add a cron that runs every 10 minutes or every hour and looks for `feedback` records created in the past `n` period. it could simply tweet them right away. so now we have a 1 hour delay (for example) for all tweets unless user pays

solution 2 kills the need for a queue at all, and while less complex may be better anyway, because realistically we'll never have 100s or even 10s of tweets upcoming. but we could have a few, and people won't want to wait.

in either case, for the queue strategy i think idea 1 will alow you to delete a couple ENV vars as well as new db attributes.

antdke commented:
@ryanckulp done!

went with option 1

1. refactored (you're right, it was messy)
2. deleted unneeded ENV vars
3. gave user a link back to the homepage so they can add more feedbacks to the queue instead of waiting on that queue page

lmk what u think
